### DISCUSSION TITLE
Document downgrade process

### DISCUSSION THREAD

**Discussion by zamazan4ik** [minio/minio#15482] [Q&A] [answered]
Hi.

After reading the documentation it is not clear for me, how the downgrade process should be performed over the solution. Could you please describe somewhere in the documentation:

- How to downgrade Minio to the older version (hopefully with no downtime). I have read that downgrade is usually supported but did not find an information, how it shall be usually performed.
- Maybe some compatibility policies and notes between versions

Thanks in advance!

**Comment by harshavardhana** [comment, marked as answer]
There is no downgrade process and its never allowed.

**Comment by zamazan4ik** [comment]
Hmm, that is a little bit confusing, since here (https://docs.min.io/minio/baremetal/installation/upgrade-minio.html) I read the following:

"Exercise particular caution if upgrading to a [release](https://github.com/minio/minio/releases) that has backwards breaking changes. MinIO includes warnings in release notes for any version known to not support downgrades.". Is it true? Does every Minio release include a warning about "we do not support downgrade"? I did not check all Minio release notes, however.

**Reply by harshavardhana** [reply]
Yes we do @zamazan4ik

**Reply by zamazan4ik** [reply]
I think it is answered now. Thanks for your time!

**Reply by ravindk89** [reply]
To clarify the docs and Harsha's comment: We do not officially support downgrades. Our docs recommend users *test first* for that reason. We take effort to mark releases where there are _definitely_ backwards breaking changes. But the absence of this warning does not indicate that the release is designed to support downgrades to some arbitrary version.

Trying to build in feature flags/gates given our rapid release schedule would add tremendous complexity to things w/ minimal returns. That kind of thing works when you have a single release every year or two, and even then it still takes a lot of user-work when it comes to upgrading and enabling feature flags/gates
